File: backend/app/models/expenditure.rb
class Expenditure
  include Mongoid::Document
  include Mongoid::Timestamps

  field :key, type: String
  field :date, type: Date
  field :motive, type: String
  field :amount, type: Float
  field :status, type: String

  embedded_in :line

  def self.total_amount
    sum(:amount)
  end
end

File: frontend/app/models/expenditure.js
import DS from 'ember-data';

export default DS.Model.extend({
  date: DS.attr('date'),
  name: DS.attr('string'),
  motive: DS.attr('string'),
  amount: DS.attr('string'),
  status: DS.attr('string'),
  key: DS.attr('string'),

  line: DS.belongsTo('line'),
});
